### PR TITLE
Feature/auctions

### PR DESCRIPTION
**Pull request by ruckim**

### CODE REVIEW

File: examples/bsc-example/src/index.ts [hunk 1]
```diff
@@ -18,6 +18,7 @@ import { bscNftExample } from './app/bsc.nft.example'
 import { bscSubscriptionsExample } from './app/bsc.subscriptions.example'
 import { bscTxWithPrivateKeyExample, bscTxWithSignatureIdExample } from './app/bsc.tx.example'
 
+// TODO add auction example
```
Comment (npwork) [inline examples/bsc-example/src/index.ts:21]: ^^

File: examples/one-example/src/index.ts [hunk 2]
```diff
@@ -16,6 +16,7 @@ import { oneOffchainExample } from './app/one.offchain.example'
 import { oneSubscriptionsExample } from './app/one.subscriptions.example'
 import { oneTxWithPrivateKeyExample, oneTxWithSignatureIdExample } from './app/one.tx.example'
 
+// TODO add ONE auction example
```
Comment (npwork) [inline examples/one-example/src/index.ts:19]: ^^

File: examples/polygon-example/src/index.ts [hunk 3]
```diff
@@ -18,6 +18,7 @@ import { polygonNftExample } from './app/polygon.nft.example'
 import { polygonSubscriptionsExample } from './app/polygon.subscriptions.example'
 import { polygonTxWithPrivateKeyExample, polygonTxWithSignatureIdExample } from './app/polygon.tx.example'
 
+// TODO add auction example
```
Comment (npwork) [inline examples/polygon-example/src/index.ts:21]: ^^

File: packages/blockchain/eth/src/lib/eth.sdk.spec.ts [hunk 4]
```diff
@@ -21,4 +22,45 @@ describe('TatumEthSDK', () => {
       walletTestFactory.generateAddressFromPrivateKey(sdk.wallet, TEST_DATA.ETH)
     })
   })
+
+  xdescribe('auction', () => {
```
Comment (npwork) [inline packages/blockchain/eth/src/lib/eth.sdk.spec.ts:26]: disabled? 
move to separate test file please

File: packages/blockchain/one/src/lib/one.sdk.ts [hunk 5]
```diff
@@ -17,6 +18,7 @@ export const TatumOneSDK = (args: SDKArguments) => {
     api,
     kms: oneKmsService({ blockchain, web3 }),
     transaction: oneTxService({ blockchain, web3 }),
+    auction: oneAuctionService({ blockchain: 'ONE' as EvmBasedBlockchain, web3 }),
```
Comment (npwork) [inline packages/blockchain/one/src/lib/one.sdk.ts:21]: why string?

File: packages/shared/testing/evm-based/src/lib/auction.test-factory.ts [hunk 6]
```diff
@@ -0,0 +1,342 @@
...
+        expect(auction).toBeDefined()
+      })
+      it('invalid address', async () => {
+        try {
+          const auction = await sdk.getAuction('invalidAddressLongerThan34Characters', '1')
+          console.log('auction', auction)
+          fail()
+        } catch (e) {
+          // TODO adjust expectation
+          expect(e.reason).toMatch('invalid address')
```
Comment (npwork) [inline packages/shared/testing/evm-based/src/lib/auction.test-factory.ts:30]: expect(...).toThrow(..) or .toThrowSdkErrorWithCode(..)

File: packages/blockchain/eth/src/lib/eth.sdk.ts [hunk 7]
```diff
@@ -22,6 +23,7 @@ export const TatumEthSDK = (args: SDKArguments) => {
       web3,
       broadcastFunction: BlockchainEthereumService.ethBroadcast,
     }),
+    auction: ethAuctionService({ blockchain, web3 }),
```
[the remark this answers is not in the document]
Comment (npwork) [reply packages/blockchain/eth/src/lib/eth.sdk.ts:26]: sounds reasonable

File: packages/blockchain/eth/src/lib/eth.sdk.spec.ts (hunk 4 above)
Comment (vlnevyhosteny) [reply packages/blockchain/eth/src/lib/eth.sdk.spec.ts:26]: done.

**Pull request merged**